packages/iov-bns/src/bnsconnection.ts

```diff
@@ -149,13 +153,42 @@ export class BnsConnection implements BcpAtomicSwapConnection {
       const { checkTx, deliverTx } = txresp;
       throw new Error(JSON.stringify({ checkTx, deliverTx }, null, 2));
     }
+    const height = txresp.height!;
+
+    const firstEvent: BcpBlockInfo = {
```

Comment:
This is a good first start (want to comment on line 151 above), but we can actually use the broadcastTxSync interface to wait for `CheckTx` on the server (made it into mempool), but not block on the commit. And then subscribe the the tx to watch when it was committed to a block.

I would also have to check if the proper message was sent on error, but I think I did this in golang a while ago, so it does actually send the info over rpc.

Reply:
I would do that in a separate PR to refine the implementation.

Reply:
Having BNS resolve after block and not after successful adding to the mempool is indeed inconsistent. It is not a huge issue because Tendermint resolves within a second or two.

> And then subscribe the the tx to watch when it was committed to a block.

Do we have code to do that (watch a transaction)? I had to use the block headers to generate my data because I did not find something.

Reply:
watchTx based on hash will get an event when it is writen to a block.

If you subscribe before the commit, you will get notified when it is inside. Then you can listen to blockheaders to increment the confirmation. 

This is just a change to the blocking call up to the first confirmation

Reply:
Thanks. Extracted to reduce the diff https://github.com/iov-one/iov-core/issues/564